### ISSUE TITLE
[Custom Image] Support CPU tags for `skypilot:` images?

### ISSUE THREAD

**Issue by romilbhardwaj** [skypilot-org/skypilot#1395]
As a new user of the custom image tags (e.g., `skypilot:gpu-ubuntu-1804`) I was a little surprised to find CPU tags not supported. Docker images (e.g., [ray](https://hub.docker.com/r/rayproject/ray/tags), [tensorflow](https://hub.docker.com/r/tensorflow/tensorflow/tags), [pytorch](https://hub.docker.com/r/pytorch/pytorch/tags)) generally support this by either supporting an explicit `cpu` tagged image or image without any tags (e.g., `skypilot:cpu-ubuntu-1804` or `skypilot:ubuntu-1804`).

I understand they might be using the same image under the hood, but good to have explicit tags for CPU. Otherwise I'm left thinking only GPU images are supported.

**Comment by jalajk24** [comment]
Can I work with this issue? I don't know a lot about how to proceed with this.

**Comment by romilbhardwaj** [comment]
Hi @jalajk24 - this issue will likely require updating the image fetcher functions for AWS to use the CPU Deep Learning AMI:
https://github.com/skypilot-org/skypilot/blob/master/sky/clouds/service_catalog/data_fetchers/fetch_aws.py#L218-L234

After this, we'll need to regenerate the catalogs and update our [catalog repo](https://github.com/skypilot-org/skypilot-catalog). 

Admittedly this is a slightly challenging issue for a new contributor, but let me know if you'd like to work on this. Otherwise there are more issues filed under the [good first issue](https://github.com/skypilot-org/skypilot/labels/good%20first%20issue) label for you to check out :)
